api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -99,6 +118,20 @@ public void setEndDate(Date endDate) {
 
 	@Override
 	public int compareTo(CohortMembership o) {
-		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
+		if (Objects.equals(this.getPatient().getPatientId(), o.getPatient().getPatientId())
```

Comment:
This method is written in a confusing way. Further, because of the use of isMembershipActive() I'm not sure it actually behaves correctly for a compareTo method.

I suggest writing it more like:

    int temp = this.getPatient().getPatientId().compareTo(o.getPatient().getPatientId());
    if (temp == 0) {
        temp = OpenmrsUtil.compareWithNullAsEarliest(this.getStartDate(), o.getStartDate());
    }
    if (temp == 0) {
        temp = OpenmrsUtil.compareWithNullAsEarliest(this.getEndDate(), o.getEndDate());
    }
    return temp;